src/pki/commands/ocsp.c

```diff
@@ -0,0 +1,569 @@
+	certificate_t *cacert = NULL, *first_issuer = NULL;
+	ocsp_request_t *ocsp_request;
+	ocsp_status_t ocsp_status = OCSP_SUCCESSFUL;
+	ocsp_responder_t *ocsp_responder = NULL;
+	linked_list_t *responses = NULL;
+	chunk_t encoding = chunk_empty, nonce = chunk_empty;
+	chunk_t issuerNameHash, issuerKeyHash, serialNumber;
+	hash_algorithm_t hashAlgorithm = HASH_SHA1, digest = HASH_UNKNOWN;
+	signature_params_t *scheme = NULL;
+	time_t lifetime = 0;
+	mem_cred_t *creds;
+	bool multiple_issuers = FALSE, self_signed = FALSE;
+	enumerator_t *enumerator;
+	int res = 1;
+
+	enum {
+		OP_SHOW,
+		OP_RESPOND,
+	} op = OP_SHOW;
+
+	bool pss = lib->settings->get_bool(lib->settings, "%s.rsa_pss", FALSE,
+									   lib->ns);
+
+	creds = mem_cred_create();
+
+	while (TRUE)
+	{
+		switch (command_getopt(&arg))
+		{
+			case 'h':
+				goto usage;
+			case 'i':
+				file = arg;
+				continue;
+			case 'r':
+				op = OP_RESPOND;
+				continue;
+			case 'k':
+				key = lib->creds->create(lib->creds,
+										 CRED_PRIVATE_KEY, KEY_ANY,
+										 BUILD_FROM_FILE, arg, BUILD_END);
+				if (!key)
+				{
+					error = "parsing private key failed";
+					goto usage;
+				}
+				creds->add_key(creds, key);
+				continue;
+			case 'c':
+				cert = lib->creds->create(lib->creds,
+										  CRED_CERTIFICATE, CERT_X509,
+										  BUILD_FROM_FILE, arg, BUILD_END);
+				if (!cert)
+				{
+					error = "parsing certificate failed";
+					goto usage;
+				}
+				creds->add_cert(creds, TRUE, cert);
+				continue;
+			case 'C':
+				cacert = lib->creds->create(lib->creds,
+											CRED_CERTIFICATE, CERT_X509,
+											BUILD_FROM_FILE, arg, BUILD_END);
+				if (!cacert)
+				{
+					error = "parsing CA certificate failed";
+					goto usage;
+				}
+				creds->add_cert(creds, TRUE, cacert);
+				continue;
+			case 'l':
+				lifetime = atoi(arg) * 60;
+				if (!lifetime)
+				{
+					error = "invalid --lifetime value";
+					goto usage;
+				}
+				continue;
+			case 'g':
+				if (!enum_from_name(hash_algorithm_short_names, arg, &digest))
+				{
+					error = "invalid --digest type";
+					goto usage;
+				}
+				continue;
+			case 'R':
+				if (!parse_rsa_padding(arg, &pss))
+				{
+					error = "invalid RSA padding";
+					goto usage;
+				}
+				continue;
+			case EOF:
+				break;
+			default:
+				error = "invalid --ocsp option";
+				goto usage;
+		}
+		break;
+	}
+
+	lib->credmgr->add_local_set(lib->credmgr, &creds->set, FALSE);
+
+	responses = linked_list_create();
+
+	if (op == OP_RESPOND && !cacert)
+	{
+		error = "respond mode requires a ca certificate";
+		goto end;
+	}
+
+	if (op == OP_RESPOND && !key)
+	{
+		error = "respond mode requires a private signer key";
+		goto end;
+	}
+
+	/* re-initialize signing certificate and key pointers */
+	cert = NULL;
+	key = NULL;
+
+	if (file)
+	{
+		ocsp_req = lib->creds->create(lib->creds, CRED_CERTIFICATE,
+								CERT_X509_OCSP_REQUEST,
+								BUILD_FROM_FILE, file, BUILD_END);
```

Comment:
These two lines could actually be indented properly (i.e. to the opening bracket), same for the `create()` call below.

Reply:
Ok, proper indentation